### DISCUSSION TITLE
DataTable row not showing all the content of cell [Row Height]

### DISCUSSION THREAD

**Discussion by MohamedElgamal** [Textualize/textual#3185] [Q&A] [answered]
Hello everyone i have a cell in the row that its data are consisting from multiple lines but the row only show one line 
the output on the screen : 
![image](https://github.com/Textualize/textual/assets/25607813/b2acc4a7-2166-4804-93c4-9f9e26cd65f8)
the details is more than one line my code snippet :
<pre>
class HistoryStateTabContent(Widget):
    ROWS = [
        ("#", "Request Type", "Details", "Creation Date")
    ]

    def __init__(self, state, username):
        super().__init__()
        self.state: StateModel = state
        self.username = username
        self.request_history_dao = request_history_dao

    def compose(self) -> ComposeResult:
        yield Label(f"{self.state.get_id()}")
        yield Label(f"{self.username}")
        yield DataTable()

    def on_mount(self):
        request_details_history = self.request_history_dao.get_requests_details_by_state(
                                        state_id=self.state.get_id(),
                                        username=self.username)
        if request_details_history is not None:
            pos = 0
            for request_detail in request_details_history:
                self.ROWS.append([pos,
                                  request_detail.get_type(),
                                  request_detail.get_request_details(),
                                  request_detail.get_created_at()])
                pos = pos + 1
        table = self.query_one(DataTable)
        table.add_columns(*self.ROWS[0])
        table.add_rows(self.ROWS[1:])

class HistoryTabContent(Widget):

    def __init__(self, username):
        super().__init__()
        self.state_dao = connection_dao
        self.username = username

    def compose(self) -> ComposeResult:
        with TabbedContent():
            for state in self.state_dao.get_states():
                with TabPane(state.get_state()):
                    with VerticalScroll():
                        yield HistoryStateTabContent(state, self.username)
</pre>

**Comment by TomJGooding** [comment, marked as answer]
You can set a height when you [add a row](https://textual.textualize.io/widgets/data_table/#textual.widgets._data_table.DataTable.add_row) to a DataTable, it  (currently) won't automatically grow to fit the content. See #3122.

**Reply by MohamedElgamal** [reply]
i tried it as following : 
        table.add_rows(cell=self.ROWS[1:], height=2)
and it give me error : 
TypeError: DataTable.add_rows() got an unexpected keyword argument 'cell'
i am new to library ,I know I am writing something wrong in function parameter but can't figure it out

**Reply by TomJGooding** [reply]
I'd recommend taking a look at the DataTable docs I linked above. `add_rows` (plural) expects just an iterable of rows, whereas `add_row` (singular) has a height parameter.
